src/hooks/useRelatedVenues.ts

```diff
@@ -57,14 +62,67 @@ const emptyArray: never[] = [];
 
 export type VenueEventsSelector = SparkleSelector<WithVenueId<VenueEvent>[]>;
 
-interface UseConnectRelatedVenuesProps {
+export interface UseRelatedVenuesProps {
+  venueId: string;
+}
+
+export interface UseRelatedVenuesData {
+  isLoading: boolean;
+  isError: boolean;
+
+  sovereignVenueId?: string;
+  isSovereignVenueIdLoading: boolean;
+  sovereignVenueIdError?: string;
+
+  relatedVenues: WithId<AnyVenue>[];
+  isRelatedVenuesLoading: boolean;
+  relatedVenuesError?: Error;
+}
+
+export const useRelatedVenues: ReactHook<
+  UseRelatedVenuesProps,
+  UseRelatedVenuesData
+> = ({ venueId }) => {
+  const {
+    sovereignVenueId,
+    isSovereignVenueIdLoading,
+    errorMsg: sovereignVenueIdError,
+  } = useSovereignVenueId({
+    venueId,
+  });
+
+  const {
+    loading: isRelatedVenuesLoading,
+    error: relatedVenuesError,
+    value: relatedVenues,
+  } = useAsync(async () => {
+    if (!sovereignVenueId) return [];
+
+    return fetchRelatedVenues(sovereignVenueId);
+  }, [sovereignVenueId]);
```

Comment:
[note] Newly implemented `useRelatedVenues`, making use of `react-use`'s `useAsync` to dramatically simplify the boilerplate required for handling async functions, _including_ properly handling error state in a way that we can simply expose it to users in the frontend.

- https://github.com/streamich/react-use
  - https://github.com/streamich/react-use/blob/master/docs/useAsync.md
  - https://github.com/streamich/react-use/blob/master/docs/useAsyncFn.md
  - https://github.com/streamich/react-use/blob/master/docs/useAsyncRetry.md